pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go

```diff
@@ -97,7 +97,11 @@ func AddOutputResources(
 		// To build copy step it needs source path(which is targetpath of input resourcemap) from task input source
 		sourcePath := inputResourceMap[boundResource.Name]
 		if sourcePath == "" {
-			sourcePath = filepath.Join(outputDir, boundResource.Name)
+			if output.TargetPath == "" {
+				sourcePath = filepath.Join(outputDir, boundResource.Name)
+			} else {
+				sourcePath = output.TargetPath
+			}
```

Comment:
could you add some unit test coverage for this?

Reply:
Sure!